pkg/api/norman/customization/cluster/validator.go

```diff
@@ -517,9 +517,6 @@ func validateEKSNodegroups(spec *v32.ClusterSpec) error {
 	if nodegroups == nil {
 		return nil
 	}
-	if len(nodegroups) == 0 {
-		return httperror.NewAPIError(httperror.InvalidBodyContent, "must have at least one nodegroup")
```

Comment:
It is requirement from Rancher. It is needed to deploy cluster-agent, otherwise downstream cluster it won't be provisioned properly.

Reply:
Even if it handles by UI? They are sending empty slice instead of nil

Reply:
@mjura I believe you're referring to this change: https://github.com/rancher/rancher/pull/43882/files. The change I’ve submitted is specifically focused on enabling the necessary UI update. It simply removes the validation that previously prevented all node pools from being removed. A corresponding update is also being made on the UI side.